query/src/main/scala/filodb/query/PlanEnums.scala

```diff
@@ -53,50 +53,60 @@ object InstantFunctionId extends Enum[InstantFunctionId] {
   // TODO time, vector, scalar
 }
 
-sealed abstract class RangeFunctionId(override val entryName: String) extends EnumEntry
+sealed trait ParamSpec
+
+case object RangeVectorParam extends ParamSpec
```

Comment:
I would suggest: add some docs, at least to the base ParamSpec trait, so we know these are used for validation.  Also, no need for blank lines between each case/variant.

Reply:
> Also, no need for blank lines between each case/variant.

I added blank lines after every cases, because in the same file for other abstract class like InstantFunctionID and more, all the cases had blank lines. 

I have fixed this blank line issue in this whole file.